deploy/charts/istio-operator/templates/operator-deployment.yaml

```diff
@@ -65,7 +65,7 @@ spec:
           {{- if .Values.clusterRegistry.clusterAPI.enabled }}
           - "--cluster-registry-api-enabled"
           {{- end }}
-          {{- if .Values.clusterRegistry.resourceSyncRules.enabled }}
+          {{- if and .Values.clusterRegistry.clusterAPI.enabled .Values.clusterRegistry.resourceSyncRules.enabled }}
```

Comment:
I deliberately separated this two config options. Most probably it will never happen, but in theory it might be that the user does not want to depend on the cluster name from the cluster CR for some reason, but uses the cluster registry controller anyway and wants to create the ResourceSyncRules.

Anyway, since I see close to zero chance on the scenario I described, I'm okay with this change, just this was my thinking.